packages/react-aria-components/src/DropZone.tsx

```diff
@@ -41,38 +42,49 @@ export interface DropZoneRenderProps {
    * Whether the dropzone is the drop target.
    * @selector [data-drop-target]
    */
-  isDropTarget: boolean
+  isDropTarget: boolean,
+  /**
+   * Whether the button is currently in a pressed state.
+   * @selector [data-pressed]
+   */
+  isPressed: boolean
 }
 // note: possibly add isDisabled prop in the future
-export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps {}
+export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps, AriaLabelingProps, PressEvents {}
 
 export const DropZoneContext = createContext<ContextValue<DropZoneProps, HTMLDivElement>>(null);
 
 function DropZone(props: DropZoneProps, ref: ForwardedRef<HTMLDivElement>) {
   [props, ref] = useContextProps(props, ref, DropZoneContext);
+  let {onPress, onPressChange, onPressEnd, onPressStart, onPressUp} = props;
   let {dropProps, isDropTarget} = useDrop({...props, ref});
   let {hoverProps, isHovered} = useHover({});
   let {focusProps, isFocused, isFocusVisible} = useFocusRing({within: true});
 
   let buttonRef = useRef<HTMLButtonElement>(null);
   let inputRef = useRef<HTMLInputElement>(null);
-  let uploadLinkRef = useRef<HTMLAnchorElement>(null);
-  let uploadButtonRef = useRef<HTMLButtonElement>(null);
   let hasButton = useHasChild('button[slot=file]', ref);
   let hasLink = useHasChild('span[slot=file]', ref);
 
+  let isMobile = useIsMobileDevice();
   let hasInput = useHasChild('input[type=file]', ref);
   let textId = useId();
   let isVirtualDrag = isVirtualDragging();
   let labelText = useMemo(() => {
-    if (!isVirtualDrag && hasInput) {
+    if (!isVirtualDrag && hasInput && !isMobile) {
       return 'Press enter to select a file';
+    } else if (!isVirtualDrag && hasInput) {
+      return 'Double tap to select a file';
```

Comment:
Mega nit: perhaps the following for more clarity on the logic flow?
```suggestion
    if (!isVirtualDrag && hasInput) {
      return isMobile ? 'Double tap to select a file' : 'Press enter to select a file';
```